projects/packages/search/src/instant-search/lib/errors.js

```diff
@@ -0,0 +1,38 @@
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Returns an error message based on the error code.
+ *
+ * @param {Error} error - Error object
+ * @returns {*}	an error message
+ */
+export function getErrorMessage( error ) {
+	switch ( error?.message ) {
+		case 'service_unavailable':
+			return __(
+				'Jetpack Search is currently unavailable. Please try again later.',
+				'jetpack-search-pkg'
+			);
+		case 'unknown_blog':
+		case 'unauthorized':
+			return __( 'You are not authorized to search on the website.', 'jetpack-search-pkg' );
```

Comment:
I'm not sure about this one. Is this something the end user would see in the overlay? If so, it seems a bit nonsensical from an end users point of view. Mostly, as a visitor to the site, I don't expect to need special permissions to run a search.

I'd probably hide the underlying reason for the error and say something like:

"Jetpack Search has encountered an error. Please contact the site administrator if the issue persists."

Reply:
To be honest, I think I'd only return two messages at the user-facing level. Those would be:

1. service_unavailable — Jetpack Search is currently unavailable. Please try again later.
2. default — Jetpack Search has encountered an error. Please contact the site administrator if the issue persists.

In my opinion, the other errors are surfacing a bit too much of the "sausage making" and would only be appropriate inside `/wp-admin` where the site admins could use the extra detail when dealing with them.

If we were to stick with these, I'd update them to the following:

1. service_unavailable — no change
2. unauthorized — Jetpack Search does not have the proper permissions to search this site. Please contact the site administrator if the issue persists.
3. bad_request — Jetpack Search was unable to process your request. Please contact the site administrator if the issue persists.
4. payload_too_large — same as bad_request
5. not_supported — Jetpack Search has encountered a configuration error. Please contact the site administrator if the issue persists.
6. default — An unknown error occurred. Please contact the site administrator if the issue persists.

From a user perspective, that's about as much info as I'd be interested in. As a site admin, it would be nice if these errors were logged or recorded and then surfaced inside **wp-admin** so I was alerted to any potential configuration issues.

Reply:
Thanks heaps for the comments. They make a lot of sense. Change incoming 😄